src/oop/object_independence/medium/q4/MainTest.java

```diff
@@ -0,0 +1,179 @@
+                        new StringLiteral("Now Playing : " + value[(current + 1) % value.length])
+                },
+                String.join(" ", "Your", firstMethod, "method does not print the correct output."));
+        assertNull(output, String.join(" ", "Your", firstMethod, "method should not return any output"));
+        _assertEquals((current + 1) % value.length, testClass.getFieldValue(classInstance, secondFieldName),
+                String.join(" ", "Your", firstMethod, "method does not increase the", secondFieldName, "attribute"));
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void portableMusicPlayerClassHasCorrectPlayPrevMethod(String[] value) throws Throwable {
+        Object[][] arguments = {
+                {value, String[].class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        String[] methodModifiers = {"public"};
+        int current = (int) testClass.getFieldValue(classInstance, secondFieldName);
+        Object output = testClass.callMethod(thirdMethod, methodModifiers, classInstance, new Clause[]{
+                        new StringLiteral("Now Playing : " + value[(current - 1 + value.length) % value.length])
+                },
+                String.join(" ", "Your", firstMethod, "method does not print the correct output."));
+        assertNull(output, String.join(" ", "Your", firstMethod, "method should not return any output"));
+        _assertEquals((current - 1 + value.length) % value.length, testClass.getFieldValue(classInstance, secondFieldName),
+                String.join(" ", "Your", firstMethod, "method does not decrease the", secondFieldName, "attribute"));
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void carClassMethodsWorkTogether(String[] value) throws Throwable {
+        Object[][] arguments = {
+                {value, String[].class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        String[] methodModifiers = {"public"};
+        int current = (int) testClass.getFieldValue(classInstance, secondFieldName);
+        Object output = testClass.callMethod(firstMethod, methodModifiers, classInstance, new Clause[]{
+                        new StringLiteral("Now Playing : " + value[current])
+                },
+                String.join(" ", "Your", firstMethod, "method does not print the correct output."));
+        assertNull(output, String.join(" ", "Your", firstMethod, "method should not return any output"));
+        _assertEquals(current, testClass.getFieldValue(classInstance, secondFieldName),
+                String.join(" ", "Your", firstMethod, "method should not alter the", secondFieldName, "attribute"));
+        for (int i = 0; i < value.length; i++) {
+            current = (int) testClass.getFieldValue(classInstance, secondFieldName);
```

Comment:
don't you already have current? Consider altering the value yourself in case the methods don't do so properly